sky/global_user_state.py

```diff
@@ -365,6 +363,18 @@ def set_cluster_status(cluster_name: str, status: ClusterStatus) -> None:
         raise ValueError(f'Cluster {cluster_name} not found.')
 
 
+def get_cluster_history_status(cluster_hash: str):
+
+    rows = _DB.cursor.execute(
+        'SELECT clusters.status FROM cluster_history '
+        'INNER JOIN clusters '
+        'ON cluster_history.cluster_hash=clusters.cluster_hash '
+        'WHERE cluster_history.cluster_hash=(?)', (cluster_hash,))
+    for (status,) in rows:
+        return status
+    return 'TERMINATED'
```

Comment:
For efficiency, why don't we join the two whole tables and retrieve the results, i.e. at the beginning of `get_clusters_from_history` do something like the following?

```
    rows = _DB.cursor.execute(
        'SELECT (ch.hash, ch.name, ch.num_nodes, ch.launched_resources, ch.usage_intervals, clusters.status)  FROM cluster_history ch'
        'INNER JOIN clusters '
        'ON ch.cluster_hash=clusters.cluster_hash '
     )
```